django_app/frontend/js/chats/streaming.js

```diff
@@ -76,19 +76,30 @@ class ChatMessage extends HTMLElement {
         let webSocket = new WebSocket(endPoint);
         let streamedContent = '';
         let sources = [];
+
+        // Stop streaming on escape key press
+        this.addEventListener('keydown', (evt) => {
+            if (evt.key === 'Escape' && this.dataset.status === 'streaming') {
+                this.dataset.status = 'stopped';
+                webSocket.close();
```

Comment:
De we need to do anything client-side to deal with this?

Reply:
Question - if the stream is escaped from, what if anything should we be saving client-side?

Reply:
It'd be great to save what the user sees, at the point of stopping it. Do you think that's a sensible approach @brunns?

Reply:
Might be - probably a question for Charlotte though?

Reply:
I'll speak with Charlotte

Reply:
In case Charlotte doesn't have time to look at it, personally I think the current/default behaviour is okay. If we get user feedback to show otherwise then we can look at it then.